airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py

```diff
@@ -76,15 +79,18 @@ def __init__(self, **data: dict[str, Any]):
 
     _dagger_client_lock: anyio.Lock = PrivateAttr(default_factory=anyio.Lock)
 
-    async def get_dagger_client(self, pipeline_name: Optional[str] = None, log_output: Optional[TextIO] = sys.stdout) -> Client:
+    async def get_dagger_client(self, pipeline_name: Optional[str] = None) -> Client:
         """
         Get (or initialize) the Dagger Client instance.
         """
         if not self._dagger_client:
             async with self._dagger_client_lock:
                 if not self._dagger_client:
+                    if self.params.get("show_dagger_logs", False):
+                        log_output = sys.stdout
+                    else:
+                        log_output, self._click_context().obj["dagger_logs_path"] = self._create_dagger_client_log_file()
```

Comment:
Nice change! 

One house keeping thing

**1. Move this logic to its own method**
```py
def get_log_output(self):
    if self.params.get("show_dagger_logs", False):
        log_output = sys.stdout
    else:
        log_output, self._click_context().obj["dagger_logs_path"] = self._create_dagger_client_log_file()

    return log_output
```